MarkUsProject/Markus
# Updated Autotest Seed Data

Updates autotest seed data so the associated test groups related `criterion_id` is appropriately referenced. 

## Motivation and Context



Currently when the autotest seed data is added to the database, the test groups that are also created do not have an associated `criterion_id`. They all currently are set to `nil` after intializing the autotest data.

Ideally these test groups would have their associated `criterion_id` populated.

## Your Changes


**Description**:

Currently when we run `rails db:autotest` to populate the autotest data, each autotest is associated with an assignment, marking scheme, submission and criterion.

Currently for each autotest schema in`db/data/autotest_files/**/*.spec.json` no criterion is specified. 

Since the `AutotestSetup` creates a criterion for each autotest based off the the assignment id we can use this to initialize the value for the autotest test group data. 

When we process the schema data for these in  https://github.com/MarkUsProject/Markus/blob/master/app/helpers/automated_tests_helper.rb#L37, we can look up the related criterion based off of the `assignment_id` and assign it as the default `criterion_id` if no criterion is provided in the seed data. 

**Type of change** (select all that apply):




## Testing



1. `docker-compose run --rm rails bash`
2. `rails db:reset`
3. `rails db:autotest`
4. Navigate to the autotest_custom assignment (under the Assignments tab), and go to Settings -> Automated Testing. This will take you to the settings page for the automated tests.
5. On that page, change the "Timeout" field from 30 to 60, and press "Save" at the bottom of the page. You should see a message at the top of the page that shows the status of updating the settings; wait until this message changes to "Completed".
6. Now go to the "Submissions" tab and click on the aaaautotest link in the leftmost column of the table. This takes you to the grading view for the submission.
7. Go to the Test Results tab and click on "Run Tests".
8. Wait a minute, and then refresh the page. Go back to the Test Results tab. You should see that two tests have been run, and that both have passed.

## Questions and Comments (if applicable)





An alternative approach might be to update the autotest seed data directly in the `db/data/autotest_files/**/*.spec.json` to include the related information for the criterion
https://github.com/MarkUsProject/Markus/blob/master/app/helpers/automated_tests_helper.rb#L20

## Checklist

- [x] I have fixed any Hound bot comments. 


### Required documentation changes (if applicable)

cc @david-yz-liu
